packages/protocol/protocol.go

```diff
@@ -212,9 +227,12 @@ func (p *Protocol) initChainManager() {
 	}, event.WithWorkerPool(wp))
 	p.Events.Engine.Consensus.SlotGadget.SlotConfirmed.Hook(func(index slot.Index) {
 		p.chainManager.CommitmentRequester.EvictUntil(index)
-	}, event.WithWorkerPool(wp))
-	p.Events.Engine.EvictionState.SlotEvicted.Hook(func(index slot.Index) {
-		p.chainManager.Evict(index)
+		newRootCommitment, err := p.Engine().Storage.Commitments.Load(index)
+		if err != nil {
+			panic(fmt.Sprintln("could not load latest confirmed commitment", err))
+		}
+		p.chainManager.SetRootCommitment(newRootCommitment)
+		p.chainManager.EvictUntil(index - 1)
```

Comment:
Can this lead to a race condition where we update the root commitment and only evict later? In this order I don't think so, just want to make sure though. If so, we should add a comment that we first need to set the rootcommitment and only then can evict.